File: nodejs/streaming_pipe_in.mjs
import * as fs from 'fs';
import * as readline from 'node:readline/promises';

const pipeIn = "input.pipe";

const input = fs.createReadStream(pipeIn);
const rl = readline.createInterface({ input })

for await (const line of rl) {
  console.log(line.toUpperCase())
}

File: java/StreamingPipeIn.java
import java.io.*;

public class StreamingPipeIn {
    public static void main(String[] args) throws Exception {
        var pipe_in = "input.pipe";

        var input = new BufferedReader(new FileReader(pipe_in));

        String line;
        while ((line = input.readLine()) != null) {
            System.out.println(line.toUpperCase());
        }

        input.close();
    }
}
